cobbler/items/item.py

```diff
@@ -151,6 +153,30 @@ def __eq__(self, other):
             return self._uid == other.uid
         return False
 
+    def __setattr__(self, name, value):
+        """
+        Intercepting an attempt to assign a value to an attribute.
+
+        :name: The attribute name.
+        :value: The attribute value.
+        """
+        if (
+            # Cache invalidation condition for to_dict.
+            # This if() has been copied from to_dict and should change when the if() in to_dict changes.
+            name.startswith("_")
+            and not name.startswith("__")
+            and name
+            not in (
+                "_conceptual_parent",
+                "_last_cached_mtime",
+                "_cached_dict",
+                "_supported_boot_loaders",
+            )
```

Comment:
If this is copied and you intend that this always is the same condition, I propose the following:

- Create a function that returns a boolean.
- The function just contains the boolean logic that you duplicated.
- The docstring of the function should describe why this Cache invalidation is working.
- The function should be private and thus start with two underscores.

Reply:
There was an additional complication with to_dict caching. For the resolved=True parameter, when changing an object, it is necessary to invalidate not just the cache of the current object, but all its children. I am currently working on the implementation.

Reply:
The cache invalidation problem turned out to be more difficult than I expected. I noticed that `System` has its own implementation of `to_dict`. Is there any reason why the `to_dict` from the `Item` base class is not used for `System`?

Reply:
If I am not mistaken you mean the to_dict method of the NetworkInterface? The reason for that is that this is a subobject of a system. The System class itself has no overwritten to_dict method.

An "easy" fix would be to promote the network interface to their own files, then I guess your logic would work but you would have a ton of other things to worry about.

Reply:
Yes, I was wrong, it is indeed the to_dict method of the NetworkInterface.

Reply:
Fixed